src/realm/set.hpp

```diff
@@ -72,6 +72,15 @@ class Set final : public CollectionBaseImpl<SetBase, Set<T>> {
         if (ndx >= current_size) {
             throw std::out_of_range("Index out of range");
         }
+
+        // proxy out the mixed value for links, we need to know if the link is valid or not (not necesseraly must it
```

Comment:
This getter is also used internally, and there are some places (eg. `Set<Mixed>::do_erase(size_t ndx)`) that do need to know if the link is unresolved or not. So we still need a private API that gives the true value.

Reply:
In those cases, we can just use 'm_tree->get(ndx)`. There must be some test missing since this is not discovered.

Reply:
Yes there is... I am looking at the code right now.

Reply:
We now use a getter internally for all the cases in which we need to know which is the underline type we are hiding behind a Mixed link.

Reply:
fixed, we use an internal getter in this  case now, which does not hide the unresolved link any more